Falcon: avoid problems with the gcc "__builtin_unreachable", replace with endless loop

--- a/arch/mips/lantiq/falcon/reset.c
+++ b/arch/mips/lantiq/falcon/reset.c
@@ -47,6 +47,10 @@ EXPORT_SYMBOL_GPL(ltq_reset_cause);
 #define WDT_PW1		0x00BE0000
 #define WDT_PW2		0x00DC0000
 
+/* to avoid problems with the gcc "__builtin_unreachable", change definition here! */
+#undef unreachable
+#define unreachable()	do { } while(1)
+
 static void machine_restart(char *command)
 {
 	local_irq_disable();
